src/modules/vtol_att_control/tailsitter.cpp

```diff
@@ -291,9 +285,18 @@ void Tailsitter::fill_actuator_outputs()
 		_actuators_out_0->control[actuator_controls_s::INDEX_THROTTLE] =
 			_actuators_fw_in->control[actuator_controls_s::INDEX_THROTTLE];
 
+		/* allow differential thrust if enabled */
+		if (_params->diff_thrust == 1) {
+			_actuators_out_0->control[actuator_controls_s::INDEX_ROLL] =
+				_actuators_fw_in->control[actuator_controls_s::INDEX_YAW] * _params->diff_thrust_scale;
+		}
+
 	} else {
 		_actuators_out_0->control[actuator_controls_s::INDEX_THROTTLE] =
 			_actuators_mc_in->control[actuator_controls_s::INDEX_THROTTLE];
+
+		_actuators_out_0->control[actuator_controls_s::INDEX_ROLL] =
+			_actuators_mc_in->control[actuator_controls_s::INDEX_ROLL] * _mc_roll_weight;
```

Comment:
Isn't this duplicating what's already done on line 277?

Reply:
it just to redefine the relations between actuators_out and actuators_in for roll axis in mc mode to avoid my code on line 290 for fw mode make roll control in mc control unpredictable. I don't know if it is necessary, so I insert this code. If you think it isn't  necessary here we can delete it but need to do fly test.

Reply:
It looks like you're repeating work already done on line 277 (`_actuators_out_0->control[actuator_controls_s::INDEX_ROLL] `).